README.md

```diff
@@ -89,6 +89,31 @@ monteprob = EnsembleProblem(prob, prob_func = prob_func, safetycopy = false)
 
 @time sol = solve(monteprob, GPUTsit5(), EnsembleGPUKernel(), trajectories = 10_000, adaptive = true, dt = 0.1f0, save_everystep = false)
 ```
+#### Callbacks with EnsembleGPUKernel
+
+Using callbacks with EnsembleGPUKernel methods requires their own GPU-compatible callback implementations. MWE:
+
+```
+function f(u, p, t)
+    du1 = -u[1]
+    return SVector{1}(du1)
+end
+
+u0 = @SVector [10.0f0]
+prob = ODEProblem{false}(f, u0, (0.0f0, 10.0f0))
+prob_func = (prob, i, repeat) -> remake(prob, p = prob.p)
+monteprob = EnsembleProblem(prob, safetycopy = false)
+
+condition(u, t, integrator) = t == 4.0f0
+affect!(integrator) = integrator.u += @SVector[10.0f0]
+
+gpu_cb = GPUDiscreteCallback(condition, affect!)
```

Comment:
wait, can't this just be done internally?

Reply:
What is meant by internally? Which part?

Reply:
Passing callback to `prob` instead of in solve?

Reply:
passing and using normal callbacks, and remaking them to GPU-specific ones right before solve.

Reply:
Yeah, that does happen. See: https://github.com/SciML/DiffEqGPU.jl/blob/master/src/DiffEqGPU.jl#L376
ATM, moreover, we are using StaticArrays, which are not CuArrays. Hence the callbacks specialization doesn't help here.
But the `DiscreteCallback` struct is not `isbits` (due save_positions being BitArray, I discussed that with you earlier), and I need to pass it to the GPU kernel.

```
gpu_cb = DiscreteCallback(condition, affect!)
sol = solve(monteprob, GPUTsit5(), EnsembleGPUKernel(),
                   trajectories = 10,
                   adaptive = false, dt = 0.01f0, callback = gpu_cb, merge_callbacks = true,
                   tstops = [4.0f0])
  Downloaded artifact: CUDA_compat
  Downloaded artifact: CUDA_compat
ERROR: GPU compilation of kernel #tsit5_kernel(CUDA.CuDeviceVector{ODEProblem{SVector{1, Float32}, Tuple{Float32, Float32}, false, SciMLBase.NullParameters, ODEFunction{false, SciMLBase.AutoSpecialize, typeof(f), LinearAlgebra.UniformScaling{Bool}, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, Nothing, typeof(SciMLBase.DEFAULT_OBSERVED), Nothing, Nothing}, Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}, SciMLBase.StandardODEProblem}, 1}, CUDA.CuDeviceMatrix{SVector{1, Float32}, 1}, CUDA.CuDeviceMatrix{Float32, 1}, Float32, CallbackSet{Tuple{}, Tuple{DiscreteCallback{typeof(condition), typeof(affect!), typeof(SciMLBase.INITIALIZE_DEFAULT), typeof(SciMLBase.FINALIZE_DEFAULT)}}}, CUDA.CuDeviceVector{Float32, 1}, Int64, Nothing, Val{true}) failed
KernelError: passing and using non-bitstype argument

Argument 6 to your kernel function is of type CallbackSet{Tuple{}, Tuple{DiscreteCallback{typeof(condition), typeof(affect!), typeof(SciMLBase.INITIALIZE_DEFAULT), typeof(SciMLBase.FINALIZE_DEFAULT)}}}, which is not isbits:
  .discrete_callbacks is of type Tuple{DiscreteCallback{typeof(condition), typeof(affect!), typeof(SciMLBase.INITIALIZE_DEFAULT), typeof(SciMLBase.FINALIZE_DEFAULT)}} which is not isbits.
    .1 is of type DiscreteCallback{typeof(condition), typeof(affect!), typeof(SciMLBase.INITIALIZE_DEFAULT), typeof(SciMLBase.FINALIZE_DEFAULT)} which is not isbits.
      .save_positions is of type BitVector which is not isbits.
        .chunks is of type Vector{UInt64} which is not isbits.
```

Reply:
Yes, but that doesn't need to impact the API. You can just convert DiscreteCallback to GPUDiscreteCallback

Reply:
Yes, but that doesn't need to impact the API. You can just convert DiscreteCallback to GPUDiscreteCallback

Reply:
I got what you meant, I will do that internally.